tests/test_gbs.py

```diff
@@ -0,0 +1,608 @@
+        )
+        a = np.ones((wires, wires))
+        params = np.ones(wires)
+
+        @qml.qnode(dev)
+        def vgbs(params):
+            ParamGraphEmbed(params, a, 1, wires=range(wires))
+            return qml.probs(wires=range(wires))
+
+        d_vgbs = qml.jacobian(vgbs, argnum=0)
+
+        p = vgbs(params)
+        dp = d_vgbs(params)
+
+        assert p.shape == (cutoff_dim ** wires,)
+        assert dp.shape == (cutoff_dim ** wires, wires)
+        assert (p >= 0).all()
+        assert (p <= 1).all()
+        assert np.sum(p) <= 1
+
+    @pytest.mark.parametrize("backend", ["gaussian", "fock"])
+    @pytest.mark.parametrize("wires", range(2, 5))
+    @pytest.mark.parametrize("cutoff_dim", [2, 3])
+    def test_shape_reduced_wires(self, wires, cutoff_dim, backend):
+        """Test that the probabilities and jacobian are returned with the expected shape when
+        probabilities are measured on a subset of wires"""
+        dev = qml.device(
+            "strawberryfields.gbs", wires=wires, cutoff_dim=cutoff_dim, backend=backend
+        )
+        a = np.ones((wires, wires))
+        params = np.ones(wires)
+
+        @qml.qnode(dev)
+        def vgbs(params):
+            ParamGraphEmbed(params, a, 1, wires=range(wires))
+            return qml.probs(wires=[0, 1])
+
+        d_vgbs = qml.jacobian(vgbs, argnum=0)
+
+        p = vgbs(params)
+        dp = d_vgbs(params)
+
+        assert p.shape == (cutoff_dim ** 2,)
+        assert dp.shape == (cutoff_dim ** 2, wires)
+        assert (p >= 0).all()
+        assert (p <= 1).all()
+        assert np.sum(p) <= 1
+
+    @pytest.mark.parametrize("wires", [range(4), Wires(["a", 42, "bob", 3])])
+    def test_example_jacobian(self, wires):
+        """Test that the jacobian is correct on the fixed example"""
+        dev = qml.device("strawberryfields.gbs", wires=wires, cutoff_dim=3)
+        params = np.array([0.25, 0.5, 0.6, 1])
+
+        @qml.qnode(dev)
+        def vgbs(params):
+            ParamGraphEmbed(params, A, 1, wires=wires)
+            return qml.probs(wires=wires)
+
+        d_vgbs = qml.jacobian(vgbs, argnum=0)
+        dp = d_vgbs(params)
+
+        assert np.allclose(dp, jac_exp)
+
+    @pytest.mark.parametrize("wires", [range(4), Wires(["a", 42, "bob", 3])])
+    @pytest.mark.parametrize("subset_wires", jac_reduced)
+    def test_example_jacobian_reduced_wires(self, subset_wires, wires):
+        """Test that the jacobian is correct on the fixed example with a subset of wires"""
+        dev = qml.device("strawberryfields.gbs", wires=wires, cutoff_dim=3)
+        params = np.array([0.25, 0.5, 0.6, 1])
+
+        @qml.qnode(dev)
+        def vgbs(params):
+            ParamGraphEmbed(params, A, 1, wires=wires)
+            return qml.probs(wires=[wires[subset_wires[0]], wires[subset_wires[1]]])
+
+        d_vgbs = qml.jacobian(vgbs, argnum=0)
+        dp = d_vgbs(params)
+
+        assert np.allclose(dp, jac_reduced[subset_wires])
```

Comment:
Really great tests!

Reply:
Yes! Very impressive and extremely thorough

Reply:
Thanks, I was worried people would fall asleep with the number of tests :laughing: